### PR TITLE
[NRH-45] Project template and Heroku setup

### PR DESCRIPTION
**Pull request by copelco**
TODO:
* [x] Create project using jade-truffle
* [x] Configure baseline Heroku Docker-based deploy
* [x] Configure web dyno
* [x] Configure celery dyno
* [x] Configure in NRH Heroku account
* [x] Configure Review Apps
* [x] Add CI for pre-commit and pytest
* [x] Setup React SPA arch
* [x] Update deployment documentation

### CODE REVIEW

File: tasks.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    print(Style.DIM + f"Tagging {tag}")
+    c.run(
+        f"docker build -t {c.config.app}:latest -t {c.config.app}:{tag} -f {dockerfile} .",
+        echo=True,
+    )
+    c.config.tag = tag
+
+
+@invoke.task
+def start_dev_server(c):
```
Comment (copelco) [inline tasks.py:55]: Does this replace `make run-dev`?

**Review by copelco — COMMENTED** [review]
Added a few comments-- looks close!

File: revengine/settings/dev.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        "whitenoise.runserver_nostatic",
+        "django.contrib.staticfiles",
+    ]
+)
+
+### React SPA index.html
+
+FRONTEND_BUILD_DIR = Path(BASE_DIR) / "spa" / "build"
+TEMPLATES[0]["DIRS"] = [FRONTEND_BUILD_DIR, os.path.join(PROJECT_DIR, "templates")]
+STATICFILES_DIRS = [os.path.join(PROJECT_DIR, "static"), str(FRONTEND_BUILD_DIR / "static")]
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply revengine/settings/dev.py:28]: I think I tried it and it wasn't working. I'll check again.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+
+```linux
+    (revengine)$ pre-commit clean
+    (revengine)$ pre-commit install
+```
+
+
+**5. Set up local env variables**
+
+This project utilizes the [direnv](https://direnv.net/) shell extension to manage project level developer environment
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply README.md:88]: I don't think we should support it anymore. It adds a dependency that we don't use in CI or CD pipelines so it's really only useful for dev, and dev is better with direnv.

File: README.md [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+```shell
+    (revengine)$ direnv allow .
+```
+
+**6. Database**
+
+The setup for local development assumes that you will be working with dockerized
+services.
+
+First add the following line to your `.env` file:
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply README.md:116]: Derp.

File: README.md [hunk 5]
```diff
@@ -0,0 +1,250 @@
...
+NOTE: The `pg_restore` command assumes that you can run `$psql` and get a prompt for your
+local database.
+
+```shell
+(revengine)$> heroku pg:backups:download --app rev-engine-test b001
+(revengine)$> pg_restore --verbose --clean --no-acl --no-owner latest.dump
+```
+
+
+## Heroku Cheatsheat
```
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply README.md:225]: It seems to make more sense here, these are actions a dev would be using in the normal course of development, not strictly in a deployment context.

File: tasks.py (hunk 1 above)
Comment (JeremyGibson) [reply tasks.py:55]: hmm. good point.

Comment (JeremyGibson) [reply tasks.py:55]: No, that was a half-formed thought. Removing it.

File: revengine/settings/dev.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JeremyGibson) [reply revengine/settings/dev.py:28]: You're right.

**Comment by copelco** [comment]
Looks good to me

**Comment by copelco** [comment]
Should the base be set to `develop`?

**Pull request merged**